sqlite.go

```diff
@@ -275,6 +275,32 @@ func (c *conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx, e
 // Raw is so ConnInitFunc can cast to SQLConn.
 func (c *conn) Raw(fn func(any) error) error { return fn(c) }
 
+// ResetSession is called prior to executing a query on the connection
+// if the connection has been used before. If the driver returns ErrBadConn
+// the connection is discarded.
+func (c *conn) ResetSession(ctx context.Context) error {
+	if c.txState != txStateNone {
+		return errors.New("cannot ResetSession on a connection with an active transaction")
+	}
+
+	// ensure that all prepared statements are reset and unbound
+	if c.stmts != nil {
+		for _, s := range c.stmts {
```

Comment:
this is almost right. in practice it would basically work. but the problem are statements created with `persist=false`, which don't get put into c.stmts.